speed up `x clean`

onur-ozkan opened the pull request:
Since `x clean` runs with bootstrap, we can speed up this process by avoiding the cleaning of bootstrap artifacts, as they are not necessarily needed to be cleaned.

ref #https://github.com/rust-lang/rust/issues/117653#issuecomment-1802482768

albertlarsan68 reviewed (commented):
Good idea

What happens with `x clean --all` on Windows ?
I do not have my main workstation on Windows anymore, so I can't test RN.

onur-ozkan commented:
> What happens with x clean --all on Windows ?

I think it will do whatever it was doing before but without purging bootstrap.

onur-ozkan commented:
> > What happens with x clean --all on Windows ?
> 
> I think it will do whatever it was doing before but without purging bootstrap.

Oh no. Wait. I am wrong actually. Let me fix that case.

onur-ozkan committed:
speed up `x clean`

Since `x clean` runs with bootstrap, we can speed up
this process by avoiding the cleaning of bootstrap artifacts,
as they are not necessarily needed to be cleaned.

Signed-off-by: onur-ozkan <work@onurozkan.dev>

onur-ozkan commented:
> Let me fix that case.

Thanks for the call! @albertlarsan68 Now `x clean` with `--all` flag should be fine on Windows.

albertlarsan68 commented:
Thanks for the PR!
@bors r+

bors commented:
:pushpin: Commit de0458af970adfd768b18bac7cb5a377f837378c has been approved by `albertlarsan68`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=albertlarsan68 de0458af970adfd768b18bac7cb5a377f837378c -->
<!-- homu: {"type":"Approved","sha":"de0458af970adfd768b18bac7cb5a377f837378c","approver":"albertlarsan68","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
